Lib/test/test_listcomps.py

```diff
@@ -163,6 +163,15 @@ def test_inner_cell_shadows_outer(self):
         outputs = {"y": [4, 4, 4, 4, 4], "i": 20}
         self._check_in_scopes(code, outputs)
 
+    def test_inner_cell_shadows_outer_no_store(self):
```

Comment:
Could you explicitly test that the value of `x` is not clobbered after the comprehension? Something like the test below, but returning both `x` and the result of the listcomp. This would guard against a possible bug where the two cells are shared and the listcomp's cell overwrites the outer function's cell.

Reply:
I think the existing `test_inner_cell_shadows_outer_redefined` test already protects against that possible bug?

Reply:
The test I've added here does return both `x` and the result of the comprehension, but it returns `x` first. I could swap the order (`return [...], x` instead of `return x, [...]`), which would give confirmation from this test also that the comprehension isn't clobbering the outer `x`?

Reply:
That one seems a little different because it contains a write to the variable in the outer scope, and that evidently makes a difference.

Reply:
The test as it looks in the PR only returns the listcomp, not x. Agree that `return [...], x` would work for what I'm asking for.

Reply:
Ah, right, I had experimented with a local change to the test, sorry :) I'll push the updated version that I described.

The presence of a write makes a difference for whether the "not treating it as a cell" previous code would be visibly broken or not by the initial `MAKE_CELL`, but now that it is treated as a cell, the presence or absence of write doesn't make any difference to the isolation. But there's certainly no harm in having both tests verify the isolation.